core/src/main/java/org/apache/iceberg/rest/RESTFileScanTaskParser.java

```diff
@@ -58,7 +58,7 @@ public static void toJson(
     generator.writeStartObject();
     generator.writeFieldName(DATA_FILE);
     ContentFileParser.toJson(fileScanTask.file(), partitionSpec, generator);
-    if (deleteFileReferences != null) {
+    if (deleteFileReferences != null && !deleteFileReferences.isEmpty()) {
```

Comment:
Thanks for fixing this, we shouldn't be producing the field if there's nothing there. But we can be more accepting on the read side as done below.